File: ios/Sources/AllowMeCapacitorPlugin/AllowMeCapacitorPlugin.swift
import Foundation
import Capacitor
import AllowMeSDK
import AllowMeSDKHomolog

@objc(AllowMeCapacitorPlugin)
public class AllowMeCapacitorPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "AllowMeCapacitorPlugin"
    public let jsName = "AllowMeCapacitor"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "initialize", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "collect", returnType: CAPPluginReturnPromise)
    ]

    private let implementation = AllowMeCapacitor()

    @objc func initialize(_ call: CAPPluginCall) {
        guard let apiKey = call.getString("apiKey"), !apiKey.isEmpty else {
            call.reject("API Key is required.")
            return
        }

        implementation.initialize(apiKey: apiKey) { result in
            switch result {
            case .success():
                call.resolve()
            case .failure(let error):
                call.reject("Initialization failed", error.localizedDescription, error)
            }
        }
    }

    @objc func collect(_ call: CAPPluginCall) {
        implementation.collect { result in
            switch result {
            case .success(let data):
                call.resolve([
                    "data": data
                ])
            case .failure(let error):
                call.reject("Collect failed", error.localizedDescription, error)
            }
        }
    }
}

File: android/src/main/java/com/mycompany/plugins/example/AllowMeCapacitorPlugin.java
package com.mycompany.plugins.example;

import android.content.Context;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.annotation.CapacitorPlugin;
import com.getcapacitor.PluginMethod;

import br.com.allowme.android.allowmesdk.SetupCallback;
import br.com.allowme.android.allowmesdk.CollectCallback;

@CapacitorPlugin(name = "AllowMeCapacitor")
public class AllowMeCapacitorPlugin extends Plugin {

    private AllowMeCapacitor allowMe;

    @PluginMethod
    public void initialize(PluginCall call) {
        String apiKey = call.getString("apiKey");

        if (apiKey == null || apiKey.isEmpty()) {
            call.reject("API Key is required.");
            return;
        }

        getActivity().runOnUiThread(() -> {
            allowMe = new AllowMeCapacitor();

            allowMe.initialize(getContext(), apiKey, new SetupCallback() {
                @Override
                public void success() {
                    call.resolve();
                }

                @Override
                public void error(Throwable throwable) {
    
                    call.reject("Initialization failed", new Exception(throwable));
                }
            });
        });
    }

    @PluginMethod
    public void collect(PluginCall call) {
        if (allowMe == null) {
            call.reject("SDK not initialized.");
            return;
        }

        allowMe.collect(new CollectCallback() {
            @Override
            public void success(String collection) {
                JSObject ret = new JSObject();
                ret.put("data", collection);
                call.resolve(ret);
            }

            @Override
            public void error(Throwable throwable) {

                call.reject("Collect failed", new Exception(throwable));
            }
        });
    }
}
